astropy/table/table.py

```diff
@@ -3712,22 +3724,32 @@ def _rows_equal(self, other):
         if isinstance(other, Table):
             other = other.as_array()
 
-        if self.has_masked_columns:
-            if isinstance(other, np.ma.MaskedArray):
-                result = self.as_array() == other
-            else:
-                # If mask is True, then by definition the row doesn't match
-                # because the other array is not masked.
-                false_mask = np.zeros(1, dtype=[(n, bool) for n in self.dtype.names])
-                result = (self.as_array().data == other) & (self.mask == false_mask)
+        self_is_masked = self.has_masked_columns
+        other_is_masked = isinstance(other, np.ma.MaskedArray)
+
+        whitelist = (TypeError, ValueError if not NUMPY_LT_1_25 else DeprecationWarning)
```

Comment:
We don't use `whitelist` now, in favor of `allowlist` / `blocklist` or other equivalents. Here you might just be more descriptive and call this `allowed_numpy_exceptions`. This will probably also have a good side effect of formatting so that it is bit more clear:
```
allowed_numpy_exceptions = (
    TypeError,
    ValueError if not NUMPY_LT_1_25 else DeprecationWarning
)